backend/import/mo_api.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (c) 2018, Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+import os
+
+from collections import defaultdict
+
+import requests
+
+from cached_property import cached_property
+
+
+DEFAULT_MO_URL = os.environ.get(
+        'MO_URL', 'http://morademo.atlas.magenta.dk/service'
+)
+ORG_ROOT = os.environ.get(
+    'MO_ORG_ROOT', '293089ba-a1d7-4fff-a9d0-79bd8bab4e5b'
+)
+
+
+def mo_get(url):
+    """Helper function for getting data from MO.
+
+    Return JSON content if successful, throw exception if not."""
+    result = requests.get(url)
+    if not result:
+        result.raise_for_status()
+    else:
+        return result.json()
+
+
+class MOData:
+    """Abstract base class to interface with MO objects."""
+
+    @cached_property
+    def json(self):
+        return mo_get(self.url)
+
+    @cached_property
+    def children(self):
+        return mo_get(self.url + '/children')
+
+    @cached_property
+    def _details(self):
+        return mo_get(self.url + '/details/')
+
+    @cached_property
+    def detail_fields(self):
+        return list(self._details.keys())
+
+    def _get_detail(self, detail):
+        return mo_get(self.url + '/details/' + detail)
+
+    def __getattr__(self, name):
```

Comment:
I'd make this `__getitem__` instead — `__getattr__` is somewhat magical.

Reply:
I like the magic. :-)

Anyway, I'm distinguishing between object members, which get stuff from MO and whose values are dictionaries, and data, which are accessed by ` __getitem__`. I like the idea of `MOData` as an object with members, and the distinction makes the code a bit cleaner, I think.